src/game/stats.c

```diff
@@ -537,3 +525,38 @@ void Stats_ShowTotal(const char *filename)
     Text_RemoveAll();
     Music_Stop();
 }
+
+void Stats_ComputeTotal(GAMEFLOW_LEVEL_TYPE level_type)
+{
+    memset(&m_TotalStats, 0, sizeof(TOTAL_STATS));
+
+    int16_t secret_flags = 0;
+
+    for (int i = 0; i < g_GameFlow.level_count; i++) {
+        if (g_GameFlow.levels[i].level_type != level_type) {
+            continue;
+        }
+        const GAME_STATS *stats = &g_GameInfo.current[i].stats;
+
+        m_TotalStats.player_kill_count += stats->kill_count;
+        m_TotalStats.player_pickup_count += stats->pickup_count;
+        secret_flags = stats->secret_flags;
+        for (int j = 0; j < MAX_SECRETS; j++) {
+            if (secret_flags & 1) {
+                m_TotalStats.player_secret_count++;
+            }
+            secret_flags >>= 1;
+        }
+
+        m_TotalStats.timer += stats->timer;
+        m_TotalStats.death_count += stats->death_count;
+        m_TotalStats.total_kill_count += stats->max_kill_count;
+        m_TotalStats.total_secret_count += stats->max_secret_count;
+        m_TotalStats.total_pickup_count += stats->max_pickup_count;
+    }
+}
```

Comment:
While I had something else on my mind (fill a struct passed as a function argument, or maybe even just return a struct), I really like this solution and I think it's even cleaner that my initial idea. Well done.

The only thing that crosses my mind is that the caller of `Stats_CheckAllSecretsCollected` needs to be aware to call `Stats_ComputeTotals(GFL_NORMAL)` first for it to work properly. We could alleviate this by making `Stats_ComputeTotal` private and calling `Stats_ComputeTotals` directly in `Stats_CheckAllSecretsCollected`.

Reply:
The reason I separated them is because `Stats_ShowTotal` needs to call `Stats_ComputeTotal`. And `Stats_ShowTotal` needs to work for both all the normal levels and all the bonus levels. And we don't necessarily want `Stats_CheckAllSecretsCollected` to be recalculated when showing total bonus level stats. I get what you are saying though, so lmk if you want it changed.

Reply:
Yeah, let's change it if possible, that way the function are not dependent on the order of calls.